tablegen/TableGen.g4

```diff
@@ -0,0 +1,481 @@
+grammar TableGen;
+
+// TableGenFile ::=  (Statement | IncludeDirective | PreprocessorDirective)*
+tableGenFile
+    :   (statement | includeDirective | preprocessorDirective)*
+    ;
```

Comment:
Please add `EOF` to the end of the rule. It forces the parser to consume the entire input, and it allows one (including the [trgen](https://github.com/kaby76/Domemtech.Trash/blob/main/src/trgen/readme.md) tool) to find the start rule.

Reply:
Thanks for the feedback. Resolved